### PR TITLE
Service: add method to define a root requirement

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Accompanies https://github.com/Homebrew/homebrew-core/pull/114915

File: Library/Homebrew/formula.rb [hunk 1]
```diff
@@ -1014,6 +1016,13 @@ def service_name
   # The generated launchd {.plist} file path.
   sig { returns(Pathname) }
   def plist_path
+    odeprecated "formula.plist_path", "formula.launchd_service_path"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula.rb:1019]: Can't add until next minor version. comment it out and add a TODO above and I'll add it then.

File: Library/Homebrew/formula.rb [hunk 2]
```diff
@@ -3121,7 +3130,10 @@ def patch(strip = :p1, src = nil, &block)
     #
     # Or perhaps you'd like to give the user a choice? Ooh fancy.
     # <pre>plist_options startup: true, manual: "foo start"</pre>
+    #
+    # @deprecated Please use {#service.require_root} instead
     def plist_options(options)
+      odeprecated "plist_options", "service.require_root"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula.rb:3136]: same

Comment (SMillerDev) [reply Library/Homebrew/formula.rb:3137]:
```suggestion
      # TODO: Deprecate
      # odeprecated "plist_options", "service.require_root"
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by Bo98** [comment]
I'm guessing we want `f.service&.requires_root? || f.plist_startup` here:

https://github.com/Homebrew/brew/blob/8684f953f621db5f1a1eed0477f34b0ee5315ed1/Library/Homebrew/caveats.rb#L181-L182

**Comment by SMillerDev** [comment]
Yeah, caveats will be a different pull request.

**Pull request merged**